### ISSUE TITLE
`opening_brace` now triggers on multiline if statements

### ISSUE THREAD

**Issue by michaeleustace** [realm/SwiftLint#5602]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

To be consistent with other multiline bracket rules, when we have a multiline if statement we put the opening brace on a new line. Before the rewrite using SwiftSyntax, this usage was never triggered. Now, however it is. 

Could this usage either be prevented from triggering the rule, or at least have a configuration option to ignore such cases.

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.55.1
* Installation method used (Homebrew, CocoaPods, building from source, etc)? Cocoapods
* Paste your configuration file:

```yml
# insert yaml contents here
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? No
  If so, paste their relative paths and respective contents.
* Which Xcode version are you using (check `xcodebuild -version`)? 15.3
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
// This does not trigger violation
guard
    condition1,
    let var1 = var1
else {
...

// This triggers a violation:
if
    condition1,
    let var1 = var1
{
```
